lib/statsd/instrument/client.rb

```diff
@@ -15,15 +15,48 @@
 #   next major release of this library. While this class may already be functional,
 #   we provide no guarantees about the API and the behavior may change.
 class StatsD::Instrument::Client
+  class << self
+    def from_env(
```

Comment:
Lightly held opinion: In the vast majority of cases you want to use the ENV values as default. e.g. you are much more likely to simply want to have a different `prefix` and / or `default_tags`, than to ignore `$STATSD_ADDR`.

As such I think `.new` should use ENV default, and that another method (no idea of a name just like that) should allow to instantiate a new client without inheriting from `ENV`.

Reply:
This is whatI had first, but changed it based on @Edouard-chin's comments. 

What I don't like about using the Environment object as default values for `initialize` is that it makes the object effective always dependent on another object. That might be dependency injection purism, but in many other instances I have been bitten by this because it increases the need for brittle mocking when writing unit tests.